go-controller/pkg/libovsdbops/datapath.go

```diff
@@ -19,20 +18,18 @@ func findDatapathByPredicate(sbClient libovsdbclient.Client, lookupFcn func(dp *
 		return nil, err
 	}
 
-	if len(datapathBindings) != 1 {
```

Comment:
@astoycos I think we should actually keep this bit here instead of in CreateMACBinding()... but change it to what you have in CreateMACBInding() so the error message is more accurate:

```
	if len(datapaths) == 0 {
		return fmt.Errorf("no datapath entries found for %s", datapathName)
	}

	if len(datapaths) > 1 {
		return fmt.Errorf("multiple datapath entries found for %s", datapathName)
	}
```

Reply:
So this was something that arose earlier in another PR regarding helper functions in libovsdbops... We basically have been putting the burden of result validation on the user of the package rather than internally (except for obvious ones like sbGlobal and nbGlobal where there should only ever be one result) what do you think @jcaamano @flavio-fernandes

Reply:
@astoycos Given that, I'm fine to move to the caller. Nevermind...